From b9624c31e72eac8aa417caf18fba01febfca7352 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 24 Aug 2021 22:48:17 +0300
Subject: [PATCH] [SW-47990] gaudi2: pll: add sram plls init

Add SRAM PLLs initialization for asic.
The SRAM PLLs initialization is needed for Preboot to be able to serve
the BMC requests in time.
For this task use SRAM PLL half nominal configuration (400MHz).

Change-Id: If404a5a26f6c3c4f713f0e540ea43dc2bf598f0d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 275a285..0b065b7 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -13,6 +13,35 @@
 #include <pll.h>
 #include <uart.h>
 
+static struct hl_pll_cfg gaudi2_pll_cfg[] = {
+	{
+		.name = "DCORE2_SRAM",
+		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 4,
+		.div_factor = { 0, -1, -1, -1 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
+		},
+	}, {
+		.name = "DCORE3_SRAM",
+		.base = mmDCORE3_XBAR_BANK_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 2, .nb = 4,
+		.div_factor = { 0, -1, -1, -1 },
+		.div_sel = { HL_PLL_PLL_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK,
+			     HL_PLL_REF_CLK
+		},
+	}
+};
+
+void hl_dev_pll_init(void)
+{
+	hl_pll_init(gaudi2_pll_cfg, ARRAY_SIZE(gaudi2_pll_cfg));
+}
+
 u64 hl_pll_timestamp_freq_get(void)
 {
 	/* As per Gaudi2 PLL cfg.xslx, CPU PLL Div[3] drives CPU timestamp */
